apps/labs/posts/pydata-theme-a11y-fixes.mdx

```diff
@@ -0,0 +1,479 @@
+following:
+
+> navigation, navigation, Section Navigation navigation, main, Breadcrumbs
+> navigation, Breadcrumb navigation, main, navigation, footer
+
+Confusing, right?
+
+![](/posts/pydata-theme-a11y-fixes/landmarks-before.png)
+
+In order to be useful, the set of landmarks needs to be short and unambiguous.
+There should be one and only one landmark for the main part of the page, just as
+there should only be one header and one footer. It’s okay to have multiple
+navigations, but they should be clear and distinct from each other.
+
+The next screenshot shows the same view with the landmarks fixed. If you read
+the following aloud, can you now make sense of the page?
+
+> Announcement complementary, banner, navigation, Section Navigation navigation,
+> main, Breadcrumb navigation, article, On this page navigation, footer
+
+Still a little bit of cleanup to be done, but much better, right?
+
+![](/posts/pydata-theme-a11y-fixes/landmarks-after.png)
+
+If you’re curious about how we did this, you can check out my [landmarks pull
+request](https://github.com/pydata/pydata-sphinx-theme/pull/1454) on GitHub.
+Here’s also a short list summarizing what I did:
+
+- Since the theme already had a `<main>` tag, I removed the `role="main"`
+  attribute from the `<article>` tag
+- Wrapped the header in a `<header>` tag (landmark = "banner")
+- Got rid of nested `<nav>`s
+- Put the announcement banner in an `<aside>` tag (landmark = "complementary")
+- Used `aria-label` and `aria-labelledby` attributes to label some of the
+  landmarks (for example, to disambiguate one navigation from another)
+
+Community feedback was key. I did not realize when I removed the `role="main"`
+attribute that [Sphinx uses the corresponding
+[role="main"]](https://github.com/sphinx-doc/sphinx/blob/4fbd3682d5cdf606159630f7daf6f818d58de05e/sphinx/themes/basic/static/searchtools.js#L185)
+CSS attribute selector to highlight search results when a reader searches a
+Sphinx-generated docs site. But one of our user-contributors, Michael Wisely,
+[reported the missing search
+highlight](https://github.com/pydata/pydata-sphinx-theme/issues/1676) on GitHub.
+Then maintainer Daniel McCloy connected the dots. The fix was fortunately
+straightforward. It just required redundantly adding the role attribute to the
+main tag, like so: `<main role="main">`. (But don't do that in your own
+day-to-day HTML! You can and should just use the `<main>` tag without the role
+attribute.)
+
+If you're curious about how I generated the screenshots in the pull request that
+I linked, or if you're curious about how your own website looks in terms of
+landmarks, you can download a browser extension like
+[WAVE](https://wave.webaim.org/extension/) or [Accessibility
+Insights](https://accessibilityinsights.io/downloads/). These extensions provide
+a way to view all of the landmarks on a web page at once.
+
+## Sidebar Table of Contents, Before and After
+
+This section assumes some knowledge of HTML and CSS.
```

Comment:
I liked that you took a moment to provide more, optional information on a screen reader in the last section, so I'm suggest equivalent links in this section. 

```suggestion
This section assumes some knowledge of [HTML](https://en.wikipedia.org/wiki/HTML) and [CSS](https://en.wikipedia.org/wiki/CSS).
```